src/lava/lib/optimization/solvers/generic/sub_process_models.py

```diff
@@ -9,22 +9,218 @@
     class NetL2:
         pass
 
+
 import numpy as np
-from lava.lib.optimization.solvers.generic.cost_integrator.process import \
-    CostIntegrator
-from lava.lib.optimization.solvers.generic.hierarchical_processes import \
-    CostConvergenceChecker, DiscreteVariablesProcess, \
-    StochasticIntegrateAndFire, NEBMAbstract, NEBMSimulatedAnnealingAbstract
-
-from lava.lib.optimization.solvers.generic.nebm.process import NEBM, \
-    NEBMSimulatedAnnealing
+from lava.lib.optimization.solvers.generic.cost_integrator.process import (
+    CostIntegrator,
+)
+from lava.lib.optimization.solvers.generic.hierarchical_processes import (
+    CostConvergenceChecker,
+    DiscreteVariablesProcess,
+    ContinuousVariablesProcess,
+    ContinuousConstraintsProcess,
+    StochasticIntegrateAndFire,
+    NEBMAbstract,
+    NEBMSimulatedAnnealingAbstract,
+)
+from lava.magma.core.resources import (
+    CPU,
+    Loihi2NeuroCore,
+    NeuroCore,
+)
+from lava.lib.optimization.solvers.generic.nebm.process import (
+    NEBM,
+    NEBMSimulatedAnnealing,
+)
 from lava.lib.optimization.solvers.generic.scif.process import QuboScif
 from lava.magma.core.decorator import implements, requires
 from lava.magma.core.model.sub.model import AbstractSubProcessModel
-from lava.magma.core.resources import Loihi2NeuroCore, CPU
 from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
 from lava.proc.dense.process import Dense
+from lava.lib.optimization.solvers.generic.qp.models import (
+    ProjectedGradientNeuronsPIPGeq,
+    ProportionalIntegralNeuronsPIPGeq,
+)
 from lava.proc.sparse.process import Sparse
+from lava.lib.optimization.utils.qp_processing import convert_to_fp
+from scipy.sparse import csr_matrix
+
+CPUS = [CPU, "CPU"]
+NEUROCORES = [Loihi2NeuroCore, NeuroCore, "Loihi2"]
+BACKEND_MSG = f""" was requested as backend. However,
+the solver currently supports only Loihi 2 and CPU backends.
+These can be specified by calling solve with any of the following:
+backend = "CPU"
+backend = "Loihi2"
+backend = CPU
+backend = Loihi2NeuroCore
+backend = NeuroCoreS
+The explicit resource classes can be imported from
+lava.magma.core.resources"""
+
+
+@implements(proc=ContinuousVariablesProcess, protocol=LoihiProtocol)
+@requires(CPU)
+class ContinuousVariablesModel(AbstractSubProcessModel):
+    """Model for the ContinuousVariables process."""
+
+    def __init__(self, proc):
+        # Instantiate child processes
+        # The input shape is a 2D vector (shape of the weight matrix).
+        backend = proc.backend
+        neuron_model = proc.hyperparameters.get("neuron_model", "qp/lp-pipg")
+
+        if neuron_model == "qp/lp-pipg":

```

Comment:
Comments in the following lines don't seem intended for release.

Reply:
They are documentation for posterity. They were intentionally left there.